src/appliance-onboarding-script/appliance_setup/avs/avsarconboarder/orchestrator/network/_network_orchestrator.py

```diff
@@ -0,0 +1,37 @@
+from ...orchestrator._orchestrator import Orchestrator
+import ipaddress
+from avs.avsarconboarder.exception import InvalidInputError
+
+class NetworkOrchestrator(Orchestrator):
+    def __init__(self):
+        self.list_of_ip_addr_in_segment = list()
+
+    def validate_static_ip_cidr_block(self, config):
+        segmant_ip_cidr = config["staticIpNetworkDetails"]["networkCIDRForApplianceVM"]
+        gateway_ip      = ""
+        char_index      = 0
+        while char_index < len(segmant_ip_cidr) and segmant_ip_cidr[char_index] != '/':
+            gateway_ip += segmant_ip_cidr[char_index]
+            char_index += 1
+
+        cidr = segmant_ip_cidr[char_index - len(segmant_ip_cidr) + 1:]
+        if int(cidr) != 28:
+            raise InvalidInputError("Invalid segamnt block size provided, Please provide a /28 address")
+        
+        try:
+            subnet_first_ip_addr = ipaddress.IPv4Address(gateway_ip) - 1
+            self.list_of_ip_addr_in_segment = list(ipaddress.ip_network('{}/{}'.format(subnet_first_ip_addr,cidr)).hosts())
```

Comment:
>ipaddress.ip_network('{}/{}'.format(subnet_first_ip_addr,cidr) [](http://example.com/codeflow?start=51&length=62)

Help me understand here why we need to reconstruct this and not use the input string as is after validation. I might be missing something

Reply:
Because the gateway address is going to be x.y.z.1. ipaddress.ip_network() won't accept such input(says host bits are set). We are thus subtracting the address by 1 to get a x.y.z.0 and use it. If the gateway is anything other than x.y.z.1, it would raise an InvalidInput exception.

Reply:
we could verify by setting strict=false.

ipaddress.IPv4Network("10.0.0.1/24", strict=False)

I know we would only be verifying if it is a valid cidr format. Whether gw ip is what we want needs to be a separate check

Reply:
your suggestion makes the code looks cleaner. Used ipaddress.IPv4Network("10.0.0.1/24", strict=False) to populate the ip addresses and also had a separate check if gateway ip is first or not